HerPublicWebsite.BusinessLogic/Services/RegularJobs/RegularJobsService.cs

```diff
@@ -45,4 +56,48 @@ public async Task RunNightlyTasksAsync()
             await dataProvider.PersistAllChangesAsync();
         }
     }
+
+    public async Task<IList<ReferralRequest>> GetReferralsPassedTenWorkingDayThresholdWithNoFollowUp()
+    {
+        var startDate = await AddWorkingDaysToDateTime(DateTime.Now, -10);
+        var newReferrals = await dataProvider.GetReferralRequestsWithNoFollowUpAfterDate(startDate);
+        return newReferrals;
+    }
+
+    public async Task<DateTime> AddWorkingDaysToDateTime(DateTime initialDateTime, int workingDaysToAdd)
+    {
+        int direction = workingDaysToAdd < 0 ? -1 : 1;
+        List<DateTime> holidays = await getHolidays();
+        DateTime newDateTime = initialDateTime;
+        while (workingDaysToAdd != 0)
+            {
+            newDateTime = newDateTime.AddDays(direction);
+            if (newDateTime.DayOfWeek != DayOfWeek.Saturday && 
+                newDateTime.DayOfWeek != DayOfWeek.Sunday && 
+                !holidays.Contains(newDateTime))
+            {
+                workingDaysToAdd -= direction;
+            }
+        }
+        return newDateTime;
+    }
+
+    private async Task<List<DateTime>> getHolidays() {
+            var parameters = new RequestParameters
+            {
+                BaseAddress = "https://www.gov.uk/bank-holidays.json",
```

Comment:
I've just realised that it won't be possible to pull out this configuration while also making the `.AddWorkingDays()` into an extension method.

The reason for that is that extension methods have to be defined in `static` classes, but configuration has to be passed in to an instance constructor.

I'll update comments elsewhere, but I think we'll want to pull out a `WorkingDaysHelper` class instead of making it an extension.

Let me know if you need help getting the dependency injection set up correctly.